tests/ops/op_math/test_exp.py

```diff
@@ -383,40 +390,147 @@ def test_non_imag_no_decomposition(self, coeff):
 
     def test_non_pauli_word_base_no_decomposition(self):
         """Tests that the decomposition doesn't exist if the base is not a pauli word."""
-        op = Exp(qml.S(0), -0.5j)
+        op = Exp(qml.S(0), -0.5j, num_steps=100)
```

Comment:
Since it doesn't have a decomposition, `num_steps` will never be used by this operator, right?

Reply:
Yes... this is not ideal. The best solution would be to use `num_steps` when calling `decomposition` (thus the `Device` should hold the information about `num_steps`). But this solution would require a lot more thought and work...